misc/chrome_extension/panel.js

```diff
@@ -13,138 +16,944 @@
-		document.getElementById('fetch').onclick = fetch;
-		document.getElementById('pretty').onclick = pretty;
-		document.getElementById('queryToAddButton').onclick = addToJson;
-		document.getElementById("queryToAddMeta").onchange = toggleField;
+	/**
+	 * Checks if the key exists in the metadata of the json
+	 * 
+	 * @param {string} key - The key
+	 * @returns {boolean} True if the key exists
+	 */
+	function DoesKeyExist(key) {
+		let textAreaElement = document.getElementById('json-config');
+		let currentValue = textAreaElement.value;
+		let json = JSON.parse(currentValue);
+		return json[0]['metadata'][key] !== undefined
+	}
+
 
-		//Hides or shows the field by default
-		toggleField();
+	/**
+	 * Creates the storage file select
+	 * 
+	 * @param {requestCallback} [callback] - After the select is created and ready to be used
+	 */
+	function initStorageSelect(callback) {
+		try {
+			let select = document.getElementById('storage');
+			select.innerHTML = '<option selected disabled>Select a file to work on</option><option disabled>------------------------------</option>';
+			getStorageValues(function (results) {
+				results.forEach(function (element) {
 
-		//Creates the value table
-		resetValueTable();
+					select.innerHTML += '<option value="' + element + '">' + element + '</option>"';
+				}, this);
+				select.innerHTML += '<option value="__create">Create new file...</option>';
+				if (callback) {
+					callback();
+				}
+			});
+
+		}
+		catch (err) {
+			alert(err);
+		}
+	}
+
+
+	/**
+	 * Gets all the name of the files stored in the storage
+	 * 
+	 * @param {requestCallback} callback - Passes the Object list to the callback
+	 * @returns {Object} An empty object if the callback isn't there
+	 */
+	function getStorageValues(callback) {
+		chrome.storage.local.get('__jsons', function (result) {
+			if (!result.__jsons) {
+				result.__jsons = [];
+			}
+
+			callback(result.__jsons);
 
-		//Connects to the messeger
-		var backgroundPageConnection = chrome.runtime.connect({
-			name: 'panel'
 		});
 
-		//Sets the textarea to the default json
-		document.getElementById('json-config').value = json;
+		return [];
+	}
+
+
+	/**
+	 * Determines wheter a file exists in storage
+	 * 
+	 * @param {string} value - The file to check 
+	 * @param {requestCallback} callback - Calls this with a {boolean}
+	 */
+	function storageValueExists(value, callback) {
+		if (value) {
+			getStorageValues(function (results) {
+				callback(results.includes(value));
+			});
+		}
+	}
+
+
+	/**
+	 * Called onchange of the storage select
+	 * Creates a new file when the create option is selected
+	 * 
+	 */
+	function storageOnChange() {
+		let e = document.getElementById('storage');
+		let errorElement = document.getElementById('storageError');
+		let value = e.options[e.selectedIndex].value;
+		errorElement.innerHTML = '';
+
+		getStorageValues(function (result) {
+
+			if (value == '__create') {
+				let newJsonName = window.prompt('New json name');
```

Comment:
What do you mean 'new json name' ? 
What does the json represent, the web scraper config? It should say that instead.